[WIP] Replace DHTMLX toolbar

jzigmund opened the pull request.

jzigmund commented:
@epwinchell PR on your request

martinpovolny commented:
@jzigmund : You are moving a lot of the logic down to the view partial here. I am afraid we need to go the other way round. Instead of hashes we need classes to represent parts of the toolbar and we need methods on those classes or methods that will work with those classes to contain the logic that I see in the view in your PR. Get to me when you are available to talk about it, please.

jzigmund commented:
@martinpovolny I'm supporting your idea about the classes, class methods, as you can recall we made a deal about the splitting work (you wanted to create the back-end - classes and proper method with all needed logic, I was working only on the UI part, so I've tried to use existing parts as much as I could. I'm fully at peace with removing all the new UI methods those I've created). I've done this PR on demand because @epwinchell could set proper CSS styling during my PTO.

Yes, let's get in touch for next cooperation

jzigmund commented:
@dclarizio @martinpovolny @skateman updated PR.

**Done:**
* removed dhtmlx js libs/css/images
* cleaned the app/helpers/application_helper/toolbar_builder.rb (basically I moved methods related to build toolbars to new presenter class)
* cleaned up the building methods from the "dead code" (i.e. we are going to not use XML definition for rendering toolbar)
* defined new hash object based on YAML definition of toolbar
* updated CSS styles based on new patternfly gem version (in app/views/layouts/_dhtmlxtoolbar.html.haml)
https://rawgit.com/patternfly/patternfly/master/tests/toolbar.html merged in https://github.com/ManageIQ/manageiq/pull/4752

**ToDo:**
* finish the route/url building for toolbar buttons
* write client-side JS (as far as I know @skateman is working on that)
* moar testing (till now I used for testing only few parts of the application - mostly Cloud/Providers and CI/Reports but also other views working. In some special views could missing some building methods from application_helper/toolbar_builder.rb, they should be just moved to the presenter with small refactoring)
* write rspecs

I'm a bit sad that from current point seems that the PR has competitive PR https://github.com/ManageIQ/manageiq/pull/4853 .. I don't understand to not work on one PR (although I see and understand the intension of creation the lazy rendering) .. I've spent on this a lot of hours, days, nights (in last 2 weeks) so I'm not amused with divergent ways.
